src/graphql/mirror.js

```diff
@@ -1133,6 +1133,11 @@ export class Mirror {
             case "PRIMITIVE":
               return b.field(fieldname);
             case "NODE":
+              if (field.fidelity.type === "UNFAITHFUL") {
+                throw new Error(
+                  "Handling unfaithful fields is not yet implemented"
```

Comment:
Are these two assertions reachable? At a glance, it looks like we should
never get here due to the constructor validation. If that’s correct,
then please replace the error text with something that clearly indicates
that this is an invariant violation (as in other errors in this file),
and mark the enclosing conditionals with `// istanbul ignore if` to keep
the coverage report in sync with the assertions.